fix(test): rework test

fbeauchamp opened the pull request:
sync handler
don't use file handler to  /
don't write file at the root of the remote

### Check list

> Check if done, if not relevant leave unchecked.

- [x] PR reference the relevant issue (e.g. `Fixes #007` or `See xoa-support#42`)
- [ ] if UI changes, a screenshot has been added to the PR
- [ ] documentation updated
- `CHANGELOG.unreleased.md`:
  - [x] enhancement/bug fix entry added
  - [x] list of packages to release updated (`${name} v${new version}`)
- **I have tested added/updated features** (and impacted code)
  - [x] unit tests (e.g. [`cron/parse.spec.js`](https://github.com/vatesfr/xen-orchestra/blob/b24400b21de1ebafa1099c56bac1de5c988d9202/%40xen-orchestra/cron/src/parse.spec.js))
  - [ ] if `xo-server` API changes, the corresponding test has been added to/updated on [`xo-server-test`](https://github.com/vatesfr/xen-orchestra/tree/master/packages/xo-server-test)
  - [x] at least manual testing

### Process

1. create a PR as soon as possible
1. mark it as `WiP:` (Work in Progress) if not ready to be merged
1. when you want a review, add a reviewer (and only one)
1. if necessary, update your PR, and re- add a reviewer

From [_the Four Agreements_](https://en.wikipedia.org/wiki/Don_Miguel_Ruiz#The_Four_Agreements):

1. Be impeccable with your word.
1. Don't take anything personally.
1. Don't make assumptions.
1. Always do your best.

julien-f commented on packages/vhd-lib/openVhd.js:
```diff
@@ -7,12 +7,12 @@ const { VhdFile } = require('./Vhd/VhdFile.js')
-    // if the remote is encrypted, trying to open a VhdFile will throw an assertion error before checking if the path is a directory, therefore we should try to open a VhdDirectory anyway.
-    if (e.code !== 'EISDIR' && e.code !== 'ERR_ASSERTION') {
+    // if the error is directory based , try to open as a vhd file
+    if (e.code !== 'EISDIR' && e.code !== 'ENOTDIR') {
```
1. I find it difficult to understand, reversing the condition might help
2. I think it's weird that we fallback on VhdFile if the error is `EISDIR` or `ENOTDIR` :thinking:

fbeauchamp committed:
fix(@xen-orchestra/fs): do not create metadata on non encrypted remote

this was creating file in wrong place during test or when running cli
introduced by #05161bd4df5b42e5ecfa0ae11e60c466ab2eabdf

fbeauchamp committed:
fix(vhd-cli): sync handler

fbeauchamp committed:
test: rework tests following 05161bd4df5b42e5ecfa0ae11e60c466ab2eabdf

Test of cleanVm are still failin , untill we fix the error condition of cleanVm broken vhd removing

- don't use handler to / (need root to run)
- don't create file at the root of the remote (conflict with the metadata and encryption.json)
- test more unhappy paths

fbeauchamp committed:
fix(vhd-lib): improve openVhd error handling

julien-f committed:
test(fs/abstract): use getSyncedHandler

julien-f reviewed (approved):
LGTM
